enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/config/ConfigHolder.java

```diff
@@ -510,8 +518,50 @@ private void populateJWTGeneratorConfigurations(JWTGenerator jwtGenerator) {
         } catch (EnforcerException | CertificateException | IOException e) {
             logger.error("Error in loading public cert or private key", e);
         }
+
         config.setJwtConfigurationDto(jwtConfigurationDto);
+        populateBackendJWKSConfiguration(jwtGenerator);
+    }
+
+    private void populateBackendJWKSConfiguration(JWTGenerator jwtGenerator) {
+        BackendJWKSDto backendJWKSDto = new BackendJWKSDto();
+        backendJWKSDto.setEnabled(jwtGenerator.getJwksEnabled());
+        if (jwtGenerator.getJwksEnabled()) {
+            String[] publicCertPaths = new String[jwtGenerator.getAdditionalJwksCertPathsCount()];
+            for (int i = 0; i < jwtGenerator.getAdditionalJwksCertPathsCount(); i++) {
+                publicCertPaths[i] = jwtGenerator.getAdditionalJwksCertPaths(i);
+            }
+            ArrayList<JWK> jwks = new ArrayList<>(publicCertPaths.length + 1);
+            // Public cert path provided to JWT generator
+            try {
+                jwks.add(jwkFromCertPath(jwtGenerator.getPublicCertificatePath()));
+            } catch (JOSEException | IOException | CertificateException e) {
+                logger.error("Error in loading public cert for JWKS ", e);
+            }
+            // Additional public cert paths
+            try {
+                for (String publicCertPath : publicCertPaths) {
+                    jwks.add(jwkFromCertPath(publicCertPath));
+                }
+            } catch (JOSEException | CertificateException | IOException e) {
+                logger.error("Error in loading public cert for JWKS", e);
```

Comment:
reword to `Error in loading additional public certs for JWKS` ?
to separately identify what happened.